🎉 Simplify dependency management

AnotherFoxGuy opened the pull request:
This PR removes the duplicate cmake code that was used to find the dependencies, now all the dependencies are defined in `cmake/DependenciesConfig.cmake` instead of the main `CMakeLists.txt`

Features:
* Unified the dependencies in one file, this makes it easier to see what versions are being used
* Replaced vcpkg with conan
* On-demand fallback on conan <sup>1</sup>
* Removed the pmm cmake-module

Fixes #785
Fixes #770
Fixes #834
Fixes #839 

<hr>

<sup>1</sup> Setting the cmake option `CYTOPIA_LIB_PREFERENCE` to `SYSTEM` will search the system first for dependencies, and then fall back on the Conan package if it hasn't been found

Mograbi commented:
pixelbuffer test need to be fixed
`D:\a\Cytopia\Cytopia\tests\util\PixelBuffer.cxx(202): FAILED:
71: due to unexpected exception with message:
71:   Exception thrown from __cdecl PixelBuffer::PixelBuffer<class std::array
71:   <struct RGBAColor,2>>(const class Rectangle &,const class std::array<struct
71:   RGBAColor,2> &) at D:\a\Cytopia\Cytopia\src\util\PixelBuffer.inl.hxx:9 -
71:   Tried to create a PixelBuffer with non matching Rectangle and Range. Width:
71:   2px, Height: 1px but buffer was 2px`

AnotherFoxGuy commented:
> @Mograbi  pixelbuffer test need to be fixed 

That seems to be an issue with VS 2022, I tested this branch yesterday and it succeeded: https://github.com/AnotherFoxGuy/Cytopia/runs/5278284858?check_suite_focus=true

AnotherFoxGuy commented:
I've now "fixed" the pixelbuffer test by downgrading the Windows CI to 2019

SimplyLiz commented:
@AnotherFoxGuy I have changed a lot in the master, including getting rid of pixel buffer. Could you please review your PR? sorry for the inconvenience

AnotherFoxGuy commented:
> @SimplyLiz  I have changed a lot in the master, including getting rid of pixel buffer. Could you please review your PR? sorry for the inconvenience

I've rebased and updated the PR for the current master branch

SimplyLiz commented:
I'd really like to keep vcpkg. It works well on my Mac and Conan caused a lot of problems in the past, up to a team member quitting due to not being able to build Cytopia anymore. 
Please add it back.

You could have system libraries (no package manager) as default option and add the possibility to still use vcpkg.
Imho this is the easiest way to build Cytopia and I don't want to switch.

AnotherFoxGuy commented:
You can still use vcpkg, you just have to set `CYTOPIA_LIB_PREFERENCE` to `SYSTEM` and include the vcpkg toolchain file

SimplyLiz commented:
Like: 
```
cmake -DCMAKE_TOOLCHAIN_FILE=/usr/local/share/vcpkg/scripts/buildsystems/vcpkg.cmake -DENABLE_ANGELSCRIPT=OFF -DCYTOPIA_LIB_PREFERENCE=SYSTEM ..
```
? 
This doesn't work:
<img width="1075" alt="image" src="https://user-images.githubusercontent.com/7029836/158836177-57a00ca0-f0f6-4654-8755-a475496f429a.png">

SimplyLiz commented:
Doesn't work with Conan  on my MacBook Air M1 (MacOS 12.1)
I've installed Conan with brew (brew install Conan). - version 1.46.2

`cmake -DCMAKE_BUILD_TYPE=RelWithDebInfo ../`
1.) It doesn't detect my macOS version 12.1. 
2.) Also it says `arch=x86_64` (The M1 chip is arm64)

I'm wondering if this would instantly break again when a new minor update of macOS is released?


```
-- Executing Conan:         REQUIRES sdl/2.0.20;sdl_image/2.0.5;sdl_ttf/2.0.18;zlib/1.2.11;libnoise/1.0.0;libpng/1.6.37;openal/1.19.1;vorbis/1.3.7;angelscript/2.35.0
        GENERATORS cmake_multi
        BUILD_REQUIRES

        OPTIONS
        SETTINGS build_type=RelWithDebInfo;compiler=apple-clang;compiler.version=13.1;compiler.libcxx=libc++;os.version=12.1;arch=x86_64

-- Conan: checking conan executable
-- Conan: Found program /opt/homebrew/bin/conan
-- Conan: Version found Conan version 1.46.2
-- Conan executing: /opt/homebrew/bin/conan install . --build missing --settings build_type=RelWithDebInfo --settings compiler=apple-clang --settings compiler.version=13.1 --settings compiler.libcxx=libc++ --settings os.version=12.1 --settings arch=x86_64
ERROR: Invalid setting '12.1' is not a valid 'settings.os.version' value.
Possible values are ['None', '10.6', '10.7', '10.8', '10.9', '10.10', '10.11', '10.12', '10.13', '10.14', '10.15', '11.0', '12.0', '13.0']
```

SimplyLiz commented:
I've update to macOS Monterey 12.3 and now the "os version" seems to be 13.1 and it still can't be found

```
compiler=apple-clang --settings compiler.version=13.1 --settings compiler.libcxx=libc++ --settings os.version= --settings arch=x86_64
ERROR: Invalid setting '' is not a valid 'settings.os.version' value.
Possible values are ['None', '10.6', '10.7', '10.8', '10.9', '10.10', '10.11', '10.12', '10.13', '10.14', '10.15', '11.0', '12.0', '13.0']
```

SimplyLiz commented:
I've created a bug report. I don't know if this is a Conan bug or in Cytopia...
https://github.com/conan-io/conan/issues/10896

SimplyLiz commented:
Please disable Angelscript per default like in master. It doesn't work on macOS

SimplyLiz commented:
It's working now! Please merge @AnotherFoxGuy

SimplyLiz reviewed (commented):
LGTM
